### ISSUE TITLE
Chapter 4 - Inference Error. Pinpointed error, however don't know how to solve

### ISSUE THREAD

**Issue by Ice-Citron** [nlp-with-transformers/notebooks#141]
## Information

The problem arises in chapter:

* [ ] Introduction
* [ ] Text Classification
* [ ] Transformer Anatomy
* [ HERE] Multilingual Named Entity Recognition
* [ ] Text Generation
* [ ] Summarization
* [ ] Question Answering
* [ ] Making Transformers Efficient in Production
* [ ] Dealing with Few to No Labels
* [ ] Training Transformers from Scratch
* [ ] Future Directions

## Describe the bug
The error stems from that if you run the exemplar colab code, and push it to your huggingface hub and try to run it. When trying to run inference, it outputs the error: "Can't load tokenizer using from_pretrained, please update its configuration: tokenizers.AddedToken() got multiple values for keyword argument 'special'".

## To Reproduce

1. Run the exemplar colab code (https://colab.research.google.com/github/nlp-with-transformers/notebooks/blob/main/04_multilingual-ner.ipynb) up to:
![Screenshot 2024-06-27 at 11 16 59 PM](https://github.com/nlp-with-transformers/notebooks/assets/137890674/b777b296-55d7-43aa-b9fd-1645bff84ff8)

2. Then visit the Huggingface hub once the model is trained and pushed to the hub, and use your personal inference API, and this tokenizer error can be seen:
![Screenshot 2024-06-27 at 11 17 38 PM](https://github.com/nlp-with-transformers/notebooks/assets/137890674/f56a0fd2-49ee-4abc-998e-a5deff331a63)

3. Same also apply if you try to import the model off it, the error occurs at the tokenizer stage, and more precisely I believe at "special_tokens_map.json".
![Screenshot 2024-06-27 at 11 18 28 PM](https://github.com/nlp-with-transformers/notebooks/assets/137890674/4e6eed14-21bf-4eb9-8e01-71a2753377a2)

4. However, this seems to be subvertable if I instead just pass in the special token of "mask_token" as an extra kwargs, as per recommended by GPT-4
![Screenshot 2024-06-27 at 11 20 17 PM](https://github.com/nlp-with-transformers/notebooks/assets/137890674/76c33b8d-446b-49f3-a66c-e36e06e785c8)

```Python
from transformers import AutoTokenizer, AutoModelForTokenClassification

# Manually specify special tokens if the default configuration is problematic
special_tokens_dict = {
    "mask_token": {
    "content": "<mask>",
    "single_word": False,
    "lstrip": True,
    "rstrip": False,
    "normalized": True,
    "special": True,  
    "__type": "AddedToken"
}
}
tokenizer = AutoTokenizer.from_pretrained("shng2025/xlm-roberta-base-finetuned-panx-de", use_fast=True, **special_tokens_dict)
model = AutoModelForTokenClassification.from_pretrained("shng2025/xlm-roberta-base-finetuned-panx-de") 
```

- it seems to be that the model can be imported if I just declare the "mask_token" in general. But I don't know what is causing this error in general.

## Expected behavior

I was expecting that once I fine-tuned the model by running the exemplar code and pushed it to the hub. That the model can be easily ran from the Inference API. Can also try and check my code on my personal notebook: https://colab.research.google.com/drive/1F5L_vL1o6WC3DxGWDF_g6ZPKTJ7dcmxR#scrollTo=orgQubxKVrNX

However, the same error occured when I ran it using the exemplar code directly, so I think it's likely due to some changes made with the library after this book was published causing this? it's still runnable as mentioned if I passed in "mask_token" as a **kwarg. But this is very strange, and I would love to know what's causing this error, as I am still learning, etc.

**Comment by Ice-Citron** [comment]
Very funny. Actually I managed to get it working now, **_I simply deleted "mask_token" from special_token_file and it worked._** Still not sure why it worked before and why it doesn't now. If possible could someone try and point out to me regarding the config changes, etc. But also best for this code to be changed too.

![Screenshot 2024-06-27 at 11 35 24 PM](https://github.com/nlp-with-transformers/notebooks/assets/137890674/83f6202d-1f51-4e53-a998-2f9cebdaa1fa)
![Screenshot 2024-06-27 at 11 35 35 PM](https://github.com/nlp-with-transformers/notebooks/assets/137890674/53c6f09b-8d21-4c1f-a88a-0d9f60905ed4)

- I will keep this issue as opened for now, so that people in the future can see and find solution for the exact same error.

**Comment by Ice-Citron** [comment]
Did further digging. I checked others who trained this too on HuggingFace, especially ones that were created within the last week. Those who ran in Tensorflow seems to still be able to run inference automatically. However, for people who used PyTorch like me, simillar issue were faced. Hence I believe this error is more PyTorch only for now, and will try and resolve and make a pull request soon when debugged. 
 
![Screenshot 2024-06-27 at 11 56 02 PM](https://github.com/nlp-with-transformers/notebooks/assets/137890674/da3a517b-a2ac-42d5-9ca5-9786955d56d8)

**Comment by Ice-Citron** [comment]
Sorry. I think I found the error. It's from now updating my libraries. I will try and update and retrain the models (and spend 0.6 usd of compute credits) and test this hypothesis again tmr.

**Comment by Ice-Citron** [comment]
I managed to get it to work. basically the install file on colab (idk if ur using colab) is faulty, or if you can call it that. basically it's installation requirement is installing older versions, when newer ones exists. And the ones that works are the newer libraries. You ca try and run this after running the default installation file:

```Python
#%%capture
!pip install transformers==4.41.2
!pip install datasets==2.20.0

!pip install pyarrow==16.0
!pip install requests==2.32.3

!pip install torch==2.3.0 torchvision==0.18.0 torchaudio==2.3.0

!pip install importlib-metadata

!pip install accelerate -U
```

And you can mainly just try and refer to my file too: https://colab.research.google.com/drive/1F5L_vL1o6WC3DxGWDF_g6ZPKTJ7dcmxR#scrollTo=r1SReYWcdRjZ

**Comment by Ice-Citron** [comment]
I will leave this here if anyone encounters the same bug. The bug is caused by this code block:

```Python
!git clone https://github.com/nlp-with-transformers/notebooks.git
%cd notebooks
from install import *
install_requirements()
```

Because it would install older depracated versions of libraries, and cause these bugs. My hypothesis at least.

**Comment by Ice-Citron** [comment]
To the authors, you should try and fix the requirement.txt and install.py to get them up to date. That's all for now.
